starter-code/contact-app/src/app/contact-list/contact-list.component.ts

```diff
@@ -19,6 +21,21 @@ export class ContactListComponent implements OnInit {
   addContact(){
     console.log("Add contact has been called");
     // add contact to contacts list
+    this.contacts.push(this.newContact);
     // clear inputs
+    this.newContact = {};
   }
+  
+  handleClick(form) {
+    if (form.valid) {
+      this.processing = true;
+      this.addContact();
+      this.processing = false;
```

Comment:
Very good. When you work with a database, remember to do this after receiving your response.